include/RED4ext/GpuApi/CommandListContext.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <RED4ext/DynArray.hpp>
+#include <REd4ext/CString.hpp>
+#include <d3d12.h>
+#include <wrl/client.h>
+
+namespace RED4ext
+{
+namespace GpuApi
+{
+
+enum class CommandListType
+{
+    Invalid,
+    Default,
+    CopySync,
+    CopyAsync,
+    Compute,
+    MAX
+};
+
+struct CommandListContext
+{
+    void AddPendingBarrier(const D3D12_RESOURCE_BARRIER& aBarrier);
+    void Close();
```

Comment:
I think `Close` is not right in this context. Should it be `Drop` or something similar?

I don't mind the name, but I gave it a thought.